packages/dcos-integration-test/extra/util/python_test_server.py

```diff
@@ -300,7 +305,7 @@ def start_http_server(listen_port):
 
     logging.info("HTTP server is starting, port: "
                  "{}, test-UUID: '{}'".format(listen_port, os.environ[TEST_UUID_VARNAME]))
-    test_server = HTTPServer(('', listen_port), TestHTTPRequestHandler)
+    test_server = ThreadingSimpleServer(('', listen_port), TestHTTPRequestHandler)
```

Comment:
If the first request, of any of the healthcheck connections fail, the server gets "stuck"